src/layer/x86/convolution_3x3_pack8_fp16.h

```diff
@@ -0,0 +1,1487 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2019 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+static void conv3x3s1_winograd64_transform_kernel_fp16_pack8_avx(const Mat& kernel, Mat& kernel_tm_pack8, int inch, int outch)
+{
+    // winograd63 transform kernel
+    Mat kernel_tm;
+    kernel_tm.create(8 * 8, inch, outch);
+
+    const float ktm[8][3] = {
+        {1.0f, 0.0f, 0.0f},
+        {-2.0f / 9, -2.0f / 9, -2.0f / 9},
+        {-2.0f / 9, 2.0f / 9, -2.0f / 9},
+        {1.0f / 90, 1.0f / 45, 2.0f / 45},
+        {1.0f / 90, -1.0f / 45, 2.0f / 45},
+        {1.0f / 45, 1.0f / 90, 1.0f / 180},
+        {1.0f / 45, -1.0f / 90, 1.0f / 180},
+        {0.0f, 0.0f, 1.0f}
+    };
+
+    #pragma omp parallel for
+    for (int p = 0; p < outch; p++)
+    {
+        for (int q = 0; q < inch; q++)
+        {
+            const float* kernel0 = (const float*)kernel + p * inch * 9 + q * 9;
+            float* kernel_tm0 = kernel_tm.channel(p).row(q);
+
+            // transform kernel, transposed
+            const float* k0 = kernel0;
+            const float* k1 = kernel0 + 3;
+            const float* k2 = kernel0 + 6;
+
+            // h
+            float tmp[8][3];
+            for (int i = 0; i < 8; i++)
+            {
+                tmp[i][0] = k0[0] * ktm[i][0] + k0[1] * ktm[i][1] + k0[2] * ktm[i][2];
+                tmp[i][1] = k1[0] * ktm[i][0] + k1[1] * ktm[i][1] + k1[2] * ktm[i][2];
+                tmp[i][2] = k2[0] * ktm[i][0] + k2[1] * ktm[i][1] + k2[2] * ktm[i][2];
+            }
+
+            // v
+            for (int j = 0; j < 8; j++)
+            {
+                float* tmpp = &tmp[j][0];
+
+                for (int i = 0; i < 8; i++)
+                {
+                    kernel_tm0[j * 8 + i] = tmpp[0] * ktm[i][0] + tmpp[1] * ktm[i][1] + tmpp[2] * ktm[i][2];
+                }
+            }
+        }
+    }
+    // interleave
+    // src = 64-inch-outch
+    // dst = 8b-8a-inch/8a-84-outch/8b;
+    kernel_tm_pack8.create(inch / 8, 64, (outch / 4) / 2 + (outch / 4) % 2, (size_t)2u * 64, 64);
```

Comment:
outch must be a multiple of 8 here
```(outch / 4) / 2 + (outch / 4) % 2``` to ```outch / 8```

Reply:
Done I settled on:    `kernel_tm_pack8.create(inch / 8, 64, outch / 8, (size_t)4u * 64, 64);`
which seems to work :)

Reply:
I settled on: `kernel_tm_pack8.create(inch / 8, 64, outch / 8, (size_t)4u * 64, 64);` your suggestion caused a segfault and this makes sense give the 8 lane memory layour.

Reply:
> I settled on: `kernel_tm_pack8.create(inch / 8, 64, outch / 8, (size_t)4u * 64, 64);` your suggestion caused a segfault and this makes sense give the 8 lane memory layour.

yes, that's what I mean. outch / 8 for the channel dimension size